core/migrations/17_deploy_tokenfactory.js

```diff
@@ -0,0 +1,16 @@
+const Finder = artifacts.require("Finder");
+const TokenFactory = artifacts.require("TokenFactory");
+
+const { getKeysForNetwork, deploy } = require("../../common/MigrationUtils.js");
+const { interfaceName } = require("../utils/Constants.js");
+
+module.exports = async function(deployer, network, accounts) {
+  const keys = getKeysForNetwork(network, accounts);
+
+  const { contract: tokenfactory } = await deploy(deployer, network, TokenFactory, { from: keys.deployer });
+
+  const finder = await Finder.deployed();
+  await finder.changeImplementationAddress(web3.utils.utf8ToHex(interfaceName.TokenFactory), tokenfactory.address, {
+    from: keys.deployer
```

Comment:
I think I probably mislead you earlier. The finder has traditionally been used for DVM-related addresses, and I'm not sure we want to change that now. We pulled the ManualPriceFeed contract out of the finder a while back for this exact reason. It might make more sense to just send the address of the TokenFactory through the constructor of the ExpiringMultiParty contract instead to preserve the separation between DVM contracts and financial contracts. That's the solution I'm leaning towards, at least.

I'd like to hear @ptare's thoughts on this, though.

Reply:
That seems reasonable to me. As discussed offline, we can make the TokenFactory address a parameter of the ExpiringMultipartyFactory.

Reply:
This makes sense, I will change